### PR TITLE
Add participant details to the aggregator

### PR DESCRIPTION
**Pull request by DamithaSenevirathne**
Add participant details to the aggregator.

### CODE REVIEW

File: conf/jasminegraph-server.properties [hunk 1]
```diff
@@ -58,6 +58,7 @@ org.jasminegraph.fl.weights.file=/var/tmp/jasminegraph-localstore/weights.txt
 org.jasminegraph.fl.flag.file=/var/tmp/jasminegraph-localstore/flag.txt
 org.jasminegraph.fl.aggregator=true
 org.jasminegraph.fl.org.training=true
+org.jasminegraph.fl.organization.file=/var/tmp/jasminegraph-localstore/orgs_details.txt
```
Comment (miyurud) [inline conf/jasminegraph-server.properties:61]: `orgs_details.txt` file is a server configuration file. It should be located in the `conf` folder of JasmineGraph.

File: src/server/JasmineGraphServer.cpp [hunk 2]
```diff
@@ -1994,31 +1994,53 @@ void JasmineGraphServer::initiateOrgCommunication(std::string graphID, std::stri
 
     Utils utils;
     int fl_clients = stoi(utils.getJasmineGraphProperty("org.jasminegraph.fl_clients"));
-        std::string flagPath = utils.getJasmineGraphProperty("org.jasminegraph.fl.flag.file");
-    int threadLimit = fl_clients + 1;
+    int orgs_count = stoi(utils.getJasmineGraphProperty("org.jasminegraph.fl.num.orgs"));
+    std::string flagPath = utils.getJasmineGraphProperty("org.jasminegraph.fl.flag.file");
+    int threadLimit = fl_clients+1;
```
Comment (miyurud) [inline src/server/JasmineGraphServer.cpp:1999]: Lets have space in between the `+` symbol.

File: src/server/JasmineGraphServer.cpp [hunk 3]
```diff
@@ -1994,31 +1994,53 @@ void JasmineGraphServer::initiateOrgCommunication(std::string graphID, std::stri
...
     if (utils.getJasmineGraphProperty("org.jasminegraph.fl.aggregator") == "true"){
    
         std::thread *orgAggThread = new std::thread[1];
         orgAggThread[0] = std::thread(initiateAggregator,"localhost",  stoi(workerVector[0].port), 
                                         stoi(workerVector[0].dataPort),trainingArgs,fl_clients, "1");
         orgAggThread[0].join();
+
+    }
+
+    std::ifstream file(utils.getJasmineGraphProperty("org.jasminegraph.fl.organization.file"));
```
Comment (miyurud) [inline src/server/JasmineGraphServer.cpp:2018]: What will happen if this file is not specified in the `jasimegraph-server.properties config` file?

File: src/server/JasmineGraphServer.cpp [hunk 4]
```diff
@@ -1994,31 +1994,53 @@ void JasmineGraphServer::initiateOrgCommunication(std::string graphID, std::stri
...
+    if (file.is_open()) {
+        std::string line;
+        while (std::getline(file, line)) {
+            line = utils.trim_copy(line, " \f\n\r\t\v");
+            std::vector<std::string> strArr = Utils::split(line.c_str(), '|');
+            std::string host = strArr[0].c_str();
+            int port = stoi(strArr[1].c_str());
+            std::string trainingArgs = "--graph_id " + strArr[2];
+            server_logger.log(to_string(strArr.size()), "info");
```
Comment (miyurud) [inline src/server/JasmineGraphServer.cpp:2027]: This log message does not have any meaning

File: src/server/JasmineGraphServer.cpp [hunk 5]
```diff
@@ -2035,7 +2057,8 @@ void JasmineGraphServer::initiateOrgCommunication(std::string graphID, std::stri
         threadID++;
 
     }
-   
+
+    server_logger.log("Python servers initiated", "info");
```
Comment (miyurud) [inline src/server/JasmineGraphServer.cpp:2061]: There is only one python server we create per JasmineGraph cluster right?

**Review by miyurud — CHANGES_REQUESTED** [review]
Please address the comments I put in this PR.

File: conf/jasminegraph-server.properties (hunk 1 above)
Comment (DamithaSenevirathne) [reply conf/jasminegraph-server.properties]: updated in c9a8c3488457482c9d93f024346727a896ea2015

File: src/server/JasmineGraphServer.cpp (hunk 2 above)
Comment (DamithaSenevirathne) [reply src/server/JasmineGraphServer.cpp:1999]: updated in 58678b1a64d640daa8361e0ff3620faf4205adb9

File: src/server/JasmineGraphServer.cpp (hunk 3 above)
Comment (DamithaSenevirathne) [reply src/server/JasmineGraphServer.cpp:2018]: Error handling added in 58678b1a64d640daa8361e0ff3620faf4205adb9

File: src/server/JasmineGraphServer.cpp (hunk 4 above)
Comment (DamithaSenevirathne) [reply src/server/JasmineGraphServer.cpp]: Removed in 58678b1a64d640daa8361e0ff3620faf4205adb9

Comment (DamithaSenevirathne) [reply src/server/JasmineGraphServer.cpp]: Removed in 58678b1a64d640daa8361e0ff3620faf4205adb9

File: src/server/JasmineGraphServer.cpp (hunk 5 above)
Comment (DamithaSenevirathne) [reply src/server/JasmineGraphServer.cpp]: Removed in 58678b1a64d640daa8361e0ff3620faf4205adb9

File: conf/jasminegraph-server.properties (hunk 1 above)
Comment (miyurud) [reply conf/jasminegraph-server.properties]: Done

File: src/server/JasmineGraphServer.cpp (hunk 2 above)
Comment (miyurud) [reply src/server/JasmineGraphServer.cpp:1999]: Done

File: src/server/JasmineGraphServer.cpp (hunk 3 above)
Comment (miyurud) [reply src/server/JasmineGraphServer.cpp:2018]: Done

File: src/server/JasmineGraphServer.cpp (hunk 4 above)
Comment (miyurud) [reply src/server/JasmineGraphServer.cpp]: Done

File: src/server/JasmineGraphServer.cpp (hunk 5 above)
Comment (miyurud) [reply src/server/JasmineGraphServer.cpp]: Done

**Review by miyurud — APPROVED** [review]

**Pull request merged**